generic_config_updater/generic_updater.py

```diff
@@ -18,9 +21,94 @@ def release_lock(self):
         pass
 
 class ChangeApplier:
-    def apply(self, change):
-        # TODO: Implement change applier
-        raise NotImplementedError("ChangeApplier.apply(change) is not implemented yet")
+    def __init__(self, service_metadata_file: str=None):
+        self.config_db = ConfigDBConnector()
+        self.config_db.connect()
+        service_metadata = {}
+        if service_metadata_file:
+            with open(service_metadata_file, 'r') as f:
+                service_metadata = json.load(f)
+        self.table_service = service_metadata.get('tables', {})
+        self.service_commands = service_metadata.get('services', {})
+        self.valiate_metadata()
+        self.previous_table_name = None
+
+    def valiate_metadata(self):
+        # TODO: check metadata file syntax
+        pass
+
+    def exec_command(self, cmd) -> int:
+        p = subprocess.Popen(cmd)
+        p.communicate()
+        return p.returncode
+
+    def on_table_operation(self, table_name):
+        services = self.table_service.get(table_name, {})
+        for service in services:
+            commands = self.service_commands.get(service, {})
+            restart_comand = commands.get('restart-command')
+            validate_command = commands.get('validate-commands')
+            if restart_comand:
+                rc = self.exec_command(restart_comand)
+                if rc != 0:
+                    raise GenericConfigUpdaterError(f"Restart command failed: {restart_comand}, rc={rc}")
+                rc = self.exec_command(validate_command)
```

Comment:
Validate commands do not need to be associated with restart commands, i.e. we should be able to verify the services that absorb the changes automatically.

Reply:
fixed.